pkg/client/services/redhatopenshift/mgmt/2022-04-01/redhatopenshift/models.go

```diff
@@ -0,0 +1,908 @@
+	objectMap := make(map[string]interface{})
+	return json.Marshal(objectMap)
+}
+
+// CloudError cloudError represents a cloud error.
+type CloudError struct {
+	// Error - An error response from the service.
+	Error *CloudErrorBody `json:"error,omitempty"`
+}
+
+// CloudErrorBody cloudErrorBody represents the body of a cloud error.
+type CloudErrorBody struct {
+	// Code - An identifier for the error. Codes are invariant and are intended to be consumed programmatically.
+	Code *string `json:"code,omitempty"`
+	// Message - A message describing the error, intended to be suitable for display in a user interface.
+	Message *string `json:"message,omitempty"`
+	// Target - The target of the particular error. For example, the name of the property in error.
+	Target *string `json:"target,omitempty"`
+	// Details - A list of additional details about the error.
+	Details *[]CloudErrorBody `json:"details,omitempty"`
+}
+
+// ClusterProfile clusterProfile represents a cluster profile.
+type ClusterProfile struct {
+	// PullSecret - The pull secret for the cluster.
+	PullSecret *string `json:"pullSecret,omitempty"`
+	// Domain - The domain for the cluster.
+	Domain *string `json:"domain,omitempty"`
+	// Version - The version of the cluster.
+	Version *string `json:"version,omitempty"`
+	// ResourceGroupID - The ID of the cluster resource group.
+	ResourceGroupID *string `json:"resourceGroupId,omitempty"`
+	// FipsValidatedModules - If FIPS validated crypto modules are used. Possible values include: 'FipsValidatedModulesDisabled', 'FipsValidatedModulesEnabled'
+	FipsValidatedModules FipsValidatedModules `json:"fipsValidatedModules,omitempty"`
+}
+
+// ConsoleProfile consoleProfile represents a console profile.
+type ConsoleProfile struct {
+	// URL - The URL to access the cluster console.
+	URL *string `json:"url,omitempty"`
+}
+
+// Display display represents the display details of an operation.
+type Display struct {
+	// Provider - Friendly name of the resource provider.
+	Provider *string `json:"provider,omitempty"`
+	// Resource - Resource type on which the operation is performed.
+	Resource *string `json:"resource,omitempty"`
+	// Operation - Operation type: read, write, delete, listKeys/action, etc.
+	Operation *string `json:"operation,omitempty"`
+	// Description - Friendly name of the operation.
+	Description *string `json:"description,omitempty"`
+}
+
+// IngressProfile ingressProfile represents an ingress profile.
+type IngressProfile struct {
+	// Name - The ingress profile name.
+	Name *string `json:"name,omitempty"`
+	// Visibility - Ingress visibility. Possible values include: 'Private', 'Public'
+	Visibility Visibility `json:"visibility,omitempty"`
+	// IP - The IP of the ingress.
+	IP *string `json:"ip,omitempty"`
+}
+
+// MasterProfile masterProfile represents a master profile.
+type MasterProfile struct {
+	// VMSize - The size of the master VMs.
+	VMSize *string `json:"vmSize,omitempty"`
```

Comment:
String is more flexible than type that is translated to an enum as we can change valid VMSizes without having to do an API release now because the VMSize is just a string and not an enum

Reply:
^ this